include/deal.II/grid/manifold_lib.h

```diff
@@ -283,29 +283,50 @@ class CylindricalManifold : public Manifold<dim,spacedim>
                        const double tolerance = 1e-10);
 
   /**
+   * Compute the cartesian coordinates for a point given in cylindrical
+   * coordinates.
+   */
+  virtual Point<3>
+  pull_back(const Point<spacedim> &space_point) const override;
+
+  /**
+   * Compute the cylindrical coordinates \f$(r, \phi, \lambda)\f$ for the given
+   * point where \f$r\f$ denotes the distance from the axis,
+   * \f$\phi\f$ the angle between the given point and the computed normal
+   * direction and \f$\lambda\f$ the axial position.
+   */
+  virtual Point<spacedim>
+  push_forward(const Point<3> &chart_point) const override;
+
+  /**
    * Compute new points on the CylindricalManifold. See the documentation of
    * the base class for a detailed description of what this function does.
    */
   virtual Point<spacedim>
   get_new_point(const std::vector<Point<spacedim> > &surrounding_points,
-                const std::vector<double>           &weights) const;
+                const std::vector<double>           &weights) const override;
+
+private:
+  /**
+   * Compute a vector that is orthogonal to the given one.
+   * Used for initializing the member variable normal_direction.
+   */
+  Point<spacedim> compute_normal(const Tensor<1,spacedim> &vector) const;
```

Comment:
If you only need it during initialization, could the function be made `static`? Or even be moved into an anonymous namespace in the .cc file?